components/portfolioStockList.js

```diff
@@ -12,20 +12,20 @@ export default class PortfolioStockList extends Component {
    }
 
    componentDidMount = () => {
-      if (this.props.listType === 'portfolio') {
-         this.setState({'stockList': [{ticker: 'RAD', fullName: 'Rite Aid Corporation', value: 200.53, numShares: 20, difference: 27.21},
-            {ticker: 'CAMT', fullName: 'Camtek LTD', value: 4021.21, numShares: 87, difference: -120.23},
-            {ticker: 'WMT', fullName: 'Walmart Inc', value: 142.23, numShares: 12, difference: 21.82},
-            {ticker: 'ZNGA', fullName: 'Zinga Inc', value: 6.23, numShares: 12, difference: -0.52}]});
-      }
-      else {
+      if (this.props.listType === 'watchlist') {
          this.setState({'stockList': [{ticker: 'RADW', fullName: 'Rite Aid Corporation', value: 200.53, difference: 27.21},
          {ticker: 'CAMTW', fullName: 'Camtek LTD', value: 4021.21, difference: -120.23},
          {ticker: 'WMTW', fullName: 'Walmart Inc', value: 142.23, difference: 21.82},
          {ticker: 'ZNGAW', fullName: 'Zinga Inc', value: 6.23, difference: -0.52}]})
       }
    }
 
+   static getDerivedStateFromProps = (props) => {
+      return {
```

Comment:
Might be a good idea to check to see if stockList in props changed when assigning it. Otherwise it will be assigning it even when there is no change.

Reply:
Done!